Makefile

```diff
@@ -62,19 +66,19 @@ docker-gen-content:
 container-gen-content: ## Generates content from external sources within a container (equiv to gen-content).
 	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hack/gen-content.sh
 
-docker-render:
+docker-render: dependencies
 	@echo -e "**** The use of docker-render is deprecated. Use container-render instead. ****" 1>&2
 	$(MAKE) container-render
 
-container-render: ## Build the site using Hugo within a container (equiv to render).
+container-render: dependencies ## Build the site using Hugo within a container (equiv to render).
 	git submodule update --init --recursive --depth 1
 	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hugo --verbose --ignoreCache --minify
 
-docker-server:
+docker-server: dependencies
 	@echo -e "**** The use of docker-server is deprecated. Use container-server instead. ****" 1>&2
 	$(MAKE) container-server
 
-container-server: ## Run Hugo locally within a container, available at http://localhost:1313/
+container-server: dependencies ## Run Hugo locally within a container, available at http://localhost:1313/
```

Comment:
This is actually:
```diff
-container-server: dependencies ## Run Hugo locally within a container, available at http://localhost:1313/
+container-server: dependencies ## Run npm locally on your host computer; then run Hugo locally within a container, available at http://localhost:1313/
```

I'd prefer to see the approach from k/website where NPM doesn't run locally; one of the benefits of a container is isolating less trusted code, whereas this approach executes it on the host.

Reply:
I have done it this way as other non-container targets would need to do this on the host anyway. Moreover, even `container-server` executes `git submodule update --init --recursive --depth 1` in the host which too could be done in the container. I am following that style in this PR.

This will need to be reworked anyway when we migrate from submodules to npm modules and I suggest that this change be done then.

Reply:
We can assume that people trust `git`, but we should minimize the other code they need to trust.

I'm really reluctant to go the other way and widen what contributors must trust.

Reply:
@sftim I have made the changes. This PR now also closes #557 and all `npm` commands run inside the container and not on the host.

Reply:
My only question now is for the `render` and `server` commands. Since these 2 commands run on the host anyway, do you think it's ok to add `npm ci` to these commands? Or should the fact that `npm ci` needs to be run manually be added to the docs and we leave it up to the folks who clone this repo to run `npm` commands?

Reply:
Yes, those are fine to run on the host.